bridge/client/webrtc.js

```diff
@@ -1365,32 +1367,60 @@
             });
         });
 
+        function getDataLength(data) {
+            if (data instanceof Blob)
+                return data.size;
+
+            if (data instanceof ArrayBuffer || ArrayBuffer.isView(data))
+                return (new Uint8Array(data)).byteLength;
+
+            return unescape(encodeURIComponent(data)).length;
+        }
+
         function encodeArrayBufferArgument(buffer) {
-            var binary = '';
+            var binary = "";
             var bufferView = new Uint8Array(buffer);
-            for (var i = 0, len = bufferView.byteLength; i < len; i++) {
+            for (var i = 0, len = bufferView.byteLength; i < len; i++)
                 binary += String.fromCharCode(bufferView[i]);
+            return binary;
```

Comment:
The escaping adds a lot though, if you try with all numbers from 0 to 255, you'll end up with a shorter json representation if you base64 encode it first. It's also quite likely that if people choose to send binary data, that data will contain a lot more zeroes than usual. Not that it makes a huge difference anyway, but I don't see the point of removing it.